### PR TITLE
Add a section on saved_model

### PR DESCRIPTION
**Pull request by MarkDaoust**
This is still in contrib, but there's no other other similarly easy way to get a saved model out, so I think it's worth including. I've included a "Caution: this is experimental" above.

Staging: https://colab.sandbox.google.com/github/MarkDaoust/docs/blob/saved-model/site/en/tutorials/keras/save_and_restore_models.ipynb

### CODE REVIEW

File: site/en/tutorials/keras/save_and_restore_models.ipynb [hunk 1]
```diff
@@ -572,11 +572,21 @@
       "source": [
         "## Save the entire model\n",
         "\n",
-        "The entire model can be saved to a file that contains the weight values, the model's configuration, and even the optimizer's configuration. This allows you to checkpoint a model and resume training later—from the exact same state—without access to the original code.\n",
+        "The entire model can be saved to a file that contains the weight values, the model's configuration, and potentially even the optimizer's configuration. This allows you to checkpoint a model and resume training later—from the exact same state—without access to the original code.\n",
```
Comment (lamberta) [inline site/en/tutorials/keras/save_and_restore_models.ipynb]:
```suggestion
        "The entire model can be saved to a file that contains the weight values, the model's configuration, and even the optimizer's configuration (depends on set up). This allows you to checkpoint a model and resume training later—from the exact same state—without access to the original code.\n",
```

File: site/en/tutorials/keras/save_and_restore_models.ipynb [hunk 2]
```diff
@@ -572,11 +572,21 @@
...
         "\n",
-        "Saving a fully-functional model in Keras is very useful—you can load them in [TensorFlow.js](https://js.tensorflow.org/tutorials/import-keras.html) and then train and run them in web browsers.\n",
+        "Saving a fully-functional model is very useful—you can load them in TensorFlow.js ([HDF5](https://js.tensorflow.org/tutorials/import-keras.html), [Saved Model](https://js.tensorflow.org/tutorials/import-saved-model.html)) and then train and run them in web browsers, or convert them to run on mobile devices using TFLite([HDF5](https://www.tensorflow.org/lite/convert/python_api#exporting_a_tfkeras_file_), [Saved Model](https://www.tensorflow.org/lite/convert/python_api#exporting_a_savedmodel_))"
```
Comment (lamberta) [inline site/en/tutorials/keras/save_and_restore_models.ipynb]:
```suggestion
        "Saving a fully-functional model is very useful—you can load them in TensorFlow.js ([HDF5](https://js.tensorflow.org/tutorials/import-keras.html), [Saved Model](https://js.tensorflow.org/tutorials/import-saved-model.html)) and then train and run them in web browsers, or convert them to run on mobile devices using TensorFlow Lite ([HDF5](https://www.tensorflow.org/lite/convert/python_api#exporting_a_tfkeras_file_), [Saved Model](https://www.tensorflow.org/lite/convert/python_api#exporting_a_savedmodel_))"
```

File: site/en/tutorials/keras/save_and_restore_models.ipynb [hunk 3]
```diff
@@ -589,6 +599,11 @@
       "source": [
         "model = create_model()\n",
         "\n",
+        "# You need to use  a keras.optimizer to restore the optimizer state from an HDF5 file. \n",
```
Comment (lamberta) [inline site/en/tutorials/keras/save_and_restore_models.ipynb]:
```suggestion
        "# You need to use a keras.optimizer to restore the optimizer state from an HDF5 file.\n",
```

File: site/en/tutorials/keras/save_and_restore_models.ipynb [hunk 4]
```diff
@@ -666,6 +677,150 @@
...
+      ]
+    },
+    {
+      "metadata": {
+        "id": "LtcN4VIb7JkK",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Caution: This method of saving a keras model is experimental, and may change in future versions. "
```
Comment (lamberta) [inline site/en/tutorials/keras/save_and_restore_models.ipynb]:
```suggestion
        "Caution: This method of saving a `tf.keras` model is experimental and may change in future versions. "
```

**Comment by MarkDaoust** [comment]
It looks like this one got buried. Thanks for the fixes @lamberta. Is there anything else we need here before a merge.

**Pull request merged**